src/Service/Core.php

```diff
@@ -155,26 +187,23 @@ public function getAppConfig($module, $language) {
    *   The language code.
    *
    * @return \Drupal\Core\Entity\EntityInterface|null
-   *   The relevant applanding node.
-   *
    * @throws \Drupal\Component\Plugin\Exception\InvalidPluginDefinitionException
    * @throws \Drupal\Component\Plugin\Exception\PluginNotFoundException
    */
   public function getAppNode($module, $language) {
     // TODO: dependency injection.
-    // TODO: prevent more than one node per language being created for each app.
-    $query = \Drupal::entityQuery('node')
+    $node_storage = $node = $this->entityTypeManager->getStorage('node');
+
+    $query = $node_storage->getQuery()
       ->condition('type', 'applanding')
       ->condition('field_spalp_app_id', $module);
-
     $nids = $query->execute();
 
+    // TODO: prevent more than one node per language being created for each app.
     $nid = end($nids);
-    $node_storage = \Drupal::entityTypeManager()->getStorage('node');
     $node = $node_storage->load($nid);
-    $node_details = $node->getTranslation($language);
```

Comment:
We will need to retain translation code from 8.x1.x

Reply:
done - needed to wrap it in a try/catch in case there is no translation